autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/CourseSelectCard.tsx

```diff
@@ -1,35 +1,202 @@
 import * as React from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import ExpandedCourseCard from './ExpandedCourseCard/ExpandedCourseCard';
-import CollapsedCourseCard from './CollapsedCourseCard/CollapsedCourseCard';
+import RemoveIcon from '@material-ui/icons/Delete';
+import CollapseIcon from '@material-ui/icons/ExpandLess';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import {
+  TextField, ButtonGroup, Button, FormLabel, Card, Typography, Collapse,
+} from '@material-ui/core';
 import { RootState } from '../../../../redux/reducer';
 import { updateCourseCard } from '../../../../redux/actions/courseCards';
+import { CustomizationLevel, CourseCardOptions } from '../../../../types/CourseCardOptions';
+import SmallFastProgress from '../../../SmallFastProgress';
+
+import * as styles from './ExpandedCourseCard/ExpandedCourseCard.css';
+import SectionSelect from './ExpandedCourseCard/SectionSelect/SectionSelect';
+import BasicSelect from './ExpandedCourseCard/BasicSelect/BasicSelect';
 
 interface CourseSelectCardProps {
   id: number;
+  collapsed: boolean;
+  shouldAnimate?: boolean;
+  removeCourseCard?: (index: number) => void;
+  resetAnimCb?: () => void;
```

Comment:
Can you add comments explaining what these are + what they're used for? (e.g. `collapsed` for memoization)